app/os_js.go

```diff
@@ -162,6 +176,24 @@ func (w *window) cleanup() {
 }
 
 func (w *window) addEventListeners() {
+	w.addEventListener(w.cnv, "webglcontextlost", func(this js.Value, args []js.Value) interface{} {
+		args[0].Call("preventDefault")
+		w.contextStatus = contextStatusLost
+		w.SetAnimating(false)
+		w.w.Event(system.StageEvent{Stage: system.StagePaused})
+		return nil
+	})
+	w.addEventListener(w.cnv, "webglcontextrestored", func(this js.Value, args []js.Value) interface{} {
+		w.contextStatus = contextStatusRestored
+		w.SetAnimating(true)
+		// Resize is required to force update the canvas content when restored.
+		w.cnv.Set("width", 0)
+		w.cnv.Set("height", 0)
+		w.resize()
+		w.w.Event(system.StageEvent{Stage: system.StageRunning})
+		w.chanRedraw <- struct{}{}
```

Comment:
This is unrelated to this change, but I worry that these unconditional channel sends risks deadlock. Why not

```
select {
case w.chanRedraw <- struct{}{}
default:
}
```

for chanRedraw and chanAnimation?

Reply:
I don't think this is resolved. You did implement non-blocking redraw for the `resize` event below, so maybe add a new method for requesting redraw?